src/hotspot/share/opto/predicates.cpp

```diff
@@ -211,11 +213,41 @@ TemplateAssertionPredicate TemplateAssertionPredicate::clone_and_replace_opaque_
   return cloned_template_assertion_predicate;
 }
 
+// This class is used to replace the input to OpaqueLoopStrideNode with a new node while leaving the other nodes
+// unchanged.
+class ReplaceOpaqueStrideInput : public BFSActions {
```

Comment:
Moved this up because we now call it from `TemplateAssertionPredicate` and not `TempalteAssertionPredicateExpression`.